dill/_dill.py

```diff
@@ -447,6 +446,44 @@ def load_session(filename='/tmp/session.pkl', main=None, **kwds):
 
 ### End: Pickle the Interpreter
 
+class sentinel(object):
+    """
+    Create a unique sentinel object that is pickled as a constant.
+    """
+    def __init__(self, name):
+        self.name = name
+    def __repr__(self):
+        return __name__ + '.' + self.name # pragma: no cover
+    def __copy__(self):
+        return self # pragma: no cover
+    def __deepcopy__(self, memo):
+        return self # pragma: no cover
+    def __reduce__(self):
+        return self.name
+    def __reduce_ex__(self, protocol):
+        return self.name
+
+class BuiltinShim(object):
+    """
+    Refers to a shim function in dill._dill if it exists and to a builtin
+    function if it doesn't exist. This choice is made during the unpickle
+    step instead of the pickling process.
+    """
+    def __init__(self, shim_name, builtin):
+        self.shim_name = shim_name
+        self.builtin = builtin
+        self.__call__ = getattr(globals(), shim_name, builtin)
+    def __copy__(self):
+        return self # pragma: no cover
+    def __deepcopy__(self, memo):
+        return self # pragma: no cover
+    def __call__(self, *args, **kwargs):
```

Comment:
If you call the shim as a function, then it throws an error as `shim_name` is not defined.  I think in `def __call__`, you should be using `self.shim_name` and `self.builtin`.